db/migrate/20170706214243_add_team_to_sources.rb

```diff
@@ -0,0 +1,10 @@
+class AddTeamToSources < ActiveRecord::Migration
+  def change
+    add_reference :sources, :team, index: true, foreign_key: true
+    Source.find_each do |s|
+      t = s.user.teams.first
+      s.team = t
+      s.save!
```

Comment:
Same things as above - guard against users without team and try to use `update_columns` if possible.